sql/2019/08_Security/08_28.sql

```diff
@@ -0,0 +1,25 @@
+#standardSQL
+# 08_28: Groupings of "feature-policy" parsed values buckets
```

Comment:
Each feature policy header could contain multiple semi-colon delimited directives. Here's a suggested query that ranks each distinct directive:

```sql
#standardSQL
# 08_28: Groupings of "feature-policy" parsed values buckets
SELECT
  policy,
  COUNT(0) AS freq,
  ROUND(COUNT(0) * 100 / SUM(COUNT(0)) OVER (), 2) AS pct
FROM
  `httparchive.almanac.summary_requests`,
  UNNEST(REGEXP_EXTRACT_ALL(LOWER(respOtherHeaders),r'feature-policy = ([^,\r\n]+)')) AS value,
  UNNEST(SPLIT(value, ';')) AS policy
WHERE
  firstHtml
GROUP BY
  policy
ORDER BY
  freq DESC
```